twine/commands/upload.py

```diff
@@ -48,6 +50,40 @@ def skip_upload(response, skip_existing, package):
             (response.status_code == 403 and msg_403 in response.text)))
 
 
+def check_status_code(response, verbose):
+    """Additional safety net to catch response code 410 in case the
+    UploadToDeprecatedPyPIDetected exception breaks.
+
+    Also includes a check for response code 405 and prints helpful error
+    message guiding users to the right repository endpoints.
+    """
+    if response.status_code == 410 and "pypi.python.org" in response.url:
+        raise exceptions.UploadToDeprecatedPyPIDetected(
+            f"It appears you're uploading to pypi.python.org (or "
+            f"testpypi.python.org). You've received a 410 error response. "
+            f"Uploading to those sites is deprecated. The new sites are "
+            f"pypi.org and test.pypi.org. Try using {utils.DEFAULT_REPOSITORY}"
+            f" (or {utils.TEST_REPOSITORY}) to upload your packages instead. "
+            f"These are the default URLs for Twine now. More at "
+            f"https://packaging.python.org/guides/migrating-to-pypi-org/.")
+    elif response.status_code == 405 and "pypi.org" in response.url:
+        raise exceptions.InvalidPyPIUploadURL(
+            f"It appears you're trying to upload to pypi.org but have an "
+            f"invalid URL. You probably want one of these two URLs: "
+            f"{utils.DEFAULT_REPOSITORY} or {utils.TEST_REPOSITORY}. Check "
+            f"your --repository-url value.")
+    try:
```

Comment:
*Tiny* style suggestion for readability:

```suggestion

    try:
```